frontend/src/utils/fieldValidation.ts

```diff
@@ -206,34 +207,63 @@ export const createMobileValidationRules: ValidationRuleFnEmailAndMobile<
 export const createNumberValidationRules: ValidationRuleFn<NumberFieldBase> = (
   schema,
 ): RegisterOptions => {
-  const { selectedValidation, customVal } = schema.ValidationOptions
+  const { selectedValidation } = schema.ValidationOptions
+  const { selectedLengthValidation, customVal } =
+    schema.ValidationOptions.LengthValidationOptions
+  const { customMin, customMax } =
+    schema.ValidationOptions.RangeValidationOptions
 
   return {
     validate: {
       required: requiredSingleAnswerValidationFn(schema),
-      validNumber: (val?: string) => {
-        if (!val || !customVal) return true
+      validNumberLength: (val: string) => {
+        if (
+          selectedValidation !== NumberSelectedValidation.Length ||
+          !val ||
+          !customVal
+        )
+          return true
 
         const currLen = val.trim().length
 
-        switch (selectedValidation) {
-          case NumberSelectedValidation.Exact:
+        switch (selectedLengthValidation) {
+          case NumberSelectedLengthValidation.Exact:
             return (
               currLen === customVal ||
               simplur`Please enter ${customVal} digit[|s] (${currLen}/${customVal})`
             )
-          case NumberSelectedValidation.Min:
+          case NumberSelectedLengthValidation.Min:
             return (
               currLen >= customVal ||
               simplur`Please enter at least ${customVal} digit[|s] (${currLen}/${customVal})`
             )
-          case NumberSelectedValidation.Max:
+          case NumberSelectedLengthValidation.Max:
             return (
               currLen <= customVal ||
               simplur`Please enter at most ${customVal} digit[|s] (${currLen}/${customVal})`
             )
         }
       },
+      validNumberRange: (val: string) => {
+        if (selectedValidation !== NumberSelectedValidation.Range || !val)
+          return true
+
+        const numVal = Number(val)
+        const hasMinimum = !!customMin
+        const hasMaximum = !!customMax
```

Comment:
If I'm not wrong, this should work since `numVal` is also a non-negative number and there would be no combination of values where this would cause an error. Also, after the front-end changes, we would no longer allow a value of 0 for `customMin`.

Of course, we could also go with your suggestion and be more explicit with the checks, what would be your preference?

Reply:
Honestly this one is a bit of a supernit. But code eliminating multiple distinct cases simultaneously (e.g. in this case the `!!` eliminating null and 0) has bitten us before so I'm extra wary of it nowadays 😅

Generally, so that our code remains correct and readable, I would prefer if we base our implementations on the intent of the type representation. In particular, we know that for customMin and customMax, number represents the presence of a constraint and null represents the absence. Although we are not allowing 0 values for these anymore, maybe one day we might change our definitions and say we allow all integers to be entered into the number field. In that case, we would have to come back and change this function. 

The question is then, why should we be coding for the future when the feature does not exist yet? It's a reasonable question, and also why I think it's more of a supernit. There have been cases where I just add a comment to the effect of `Safe because customMin, customMax > 0` and just leave it, too. I think either is fine, but we should definitely document it somewhere

Reply:
I see. I think I would go with an explicit null check as it is more robust! (perhaps the number field would allow negative and zero values in the future?)